### PR TITLE
test: Add self-test cases in the `math/power_of_two.cpp` file

### PR DESCRIPTION
**Pull request by Rijul24**
#### Description of Change
<!
//* Added Self Implementation cases
//* Necessary changes for user inputs
//* Changed necessary comments

 
Thank you for your Pull Request. Please provide a description above and review
the requirements below.

Contributors guide: https://github.com/TheAlgorithms/C-Plus-Plus/CONTRIBUTING.md
-->

#### Checklist

- [x] Added description of change
- [ ] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [ ] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [ ] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: 

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C-Plus-Plus/pull/1640"></a>

### CODE REVIEW

**Review by Panquesito7 — APPROVED** [review]
LGTM 🚀 Thank you for your contribution! 😄👍

File: math/power_of_two.cpp [hunk 1]
```diff
@@ -19,48 +19,88 @@
...
-     */
+int power_of_two(int n) {
     /// result stores the
     /// bitwise and of n and n-1
     int result = n & (n - 1);
+    
     if (result == 0) {
-        std::cout << "Yes, the number " << n << " is a power of 2";
+        return 1;
     } else {
```
Comment (CarlosZoft) [inline math/power_of_two.cpp:49]: 'else' is redundant in this case

[the remark this answers is not in the document]
Comment (Panquesito7) [reply math/power_of_two.cpp:49]: Hey there! Want to make a PR to fix this?

[the remark this answers is not in the document]
Comment (CarlosZoft) [reply math/power_of_two.cpp:49]: yes! i can do this

**Pull request merged**